c/shrmem64.c

```diff
@@ -0,0 +1,364 @@
+
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+#ifdef METTLE
+#include <metal/metal.h>
+#include <metal/stddef.h>
+#include <metal/stdint.h>
+#else
+#include "stddef.h"
+#include "stdint.h"
+#endif
+
+#include <shrmem64.h>
+#include "zos.h"
+
+#ifndef _LP64
+#error ILP32 is not supported
+#endif
+
+typedef uint64_t MemObj;
+
+#define IARV64_V4PLIST_SIZE 160
```

Comment:
prefer "static const" to #define

Reply:
I'm not sure I want to use a static const. That value is used to define the size of an array. A static const would make the compiler use VLA which are optional in C11.